src/main/java/org/aludratest/service/jms/impl/JmsActionImpl.java

```diff
@@ -65,59 +83,37 @@
     private String password;
 
 	private String clientId;
+	
+	private String recentMessage;
+
 
 	public JmsActionImpl(ConnectionFactory connectionFactory, InitialContext context, final String userName, final String password) {
-        this.dynamicConnections = new HashMap<String, Connection>();
-        this.durableConsumers = new HashMap<String, TopicSubscriber>();
         this.connectionFactory = connectionFactory;
 		this.context = context;
         this.userName = userName;
         this.password = password;
+        this.topicHandlers = new HashMap<String, TopicHandler>();
 		this.clientId = userName + "@" + JmsActionImpl.class.getSimpleName() + this.hashCode();
+		this.recentMessage = null;
     }
 
 	public void close() {
 		LOGGER.info("Closing JmsServcie for clientId " + this.clientId );
-		for (TopicSubscriber subscriber : this.durableConsumers.values()) {
-			try {
-				subscriber.close();
-			}
-			catch (JMSException e) {
-				LOGGER.debug("Failed to close subscriber: ", e );
-			}
-		}
-		for (Map.Entry<String, Connection> entry : this.dynamicConnections.entrySet()) {
-			try {
-				Connection dynC = entry.getValue();
-				dynC.stop();
-				dynC.close();
-			}
-			catch (JMSException e) {
-				LOGGER.debug("Failed to close dynamic connection [ " + entry.getKey() + " ] : ", e );
-			}
-		}
-		if (session != null) {
-			try {
-                session.close();
-			}
-			catch (JMSException e) {
-				LOGGER.debug("Failed to close jms session : ", e );
-			}
-		}
-		if (connection != null) {
-			try {
-				connection.stop();
-				connection.close();
-			}
-			catch (JMSException e) {
-				LOGGER.debug("Failed to close jms connection for client-id [ " + this.clientId + " ] : ", e );
-			}
+		for (TopicHandler handler : this.topicHandlers.values()) {
+			handler.stop();
+			handler.close();
 		}
+		close(session);
+		stopAndClose(connection, "jms connection for client-id [ " + this.clientId + " ]");
 	}
-
+	
 	@Override
 	public List<Attachment> createDebugAttachments() {
-		return null;
+        List<Attachment> attachments = new ArrayList<Attachment>();
+        if (this.recentMessage != null) {
+            attachments.add(new StringAttachment("message content", this.recentMessage, "txt"));
+        }
+        return attachments;
```

Comment:
I'm not sure that this implementation does what you aim for... Wouldn't it attach the contents of recentMessage to EVERY service call once it is != null?